.github/workflows/release.yml

```diff
@@ -1,15 +1,24 @@
----
-name: release
+name: Release
 
 on:
   push:
     tags:
       - v*.*.*
 jobs:
   publish:
+    name: Release to RubyGems
     runs-on: ubuntu-latest
 
     steps:
+      - name: Wait for tests to succeed
+        uses: lewagon/wait-on-check-action@v1.0.0
```

Comment:
Out of curiosity, have you considered using a workflow dispatch that runs only upon the test are successful?